py/desitarget/secondary.py

```diff
@@ -576,20 +576,42 @@ def match_secondary(primtargs, scxdir, scndout, sep=1.,
     # APC obsconditions for specific DESI_TARGET bits
     # APC See https://github.com/desihub/desitarget/pull/530
 
-    # APC Only consider primary targets with secondary bits set
-    scnd_update = (targs[desicols[0]] & desi_mask['SCND_ANY']) != 0
-    if np.any(scnd_update):
-        # APC Allow changes to primaries if the DESI_TARGET bitmask has
-        # APC only the following bits set, in any combination.
-        log.info('Test if secondaries can update {} matched primaries'.format(
-            scnd_update.sum()))
-        update_from_scnd_bits = (desi_mask['SCND_ANY'] | desi_mask['MWS_ANY'] |
-                                 desi_mask['STD_BRIGHT'] | desi_mask['STD_FAINT']
-                                 | desi_mask['STD_WD'])
-        scnd_update &= ((targs[desicols[0]] & ~update_from_scnd_bits) == 0)
-        log.info('New priority, numobs, obscon for {} matched primaries'.format(
-            scnd_update.sum()))
+    # APC Default behaviour is that targets with SCND_ANY bits set will ONLY be
+    # APC have initial state set based on their secondary targetmask parameters IF
+    # APC they have NO primary target bits set (hence == on next line).
+    scnd_update = targs[desicols[0]] == desi_mask['SCND_ANY']
+    log.info('{} scnd targets will have initial state set as secondary-only'.format(scnd_update.sum()))
+
+    # APC The exception to the rule above is that a subset of bits flagged with
+    # APC updatemws=True in the targetmask can drive initial state for a subset of
+    # APC primary bits corresponding to MWS targets and standards. We first create
+    # APC a bitmask of those permitted seconday bits.
+    permit_scnd_bits = 0
+    for name in scnd_mask.names():
+        if hasattr(scnd_mask[name], 'updatemws'):
```

Comment:
I'd rather this failed if the needed attribute wasn't present, to ensure we always remember to include the `updatemws` attribute. But, I understand that if you remove this unit tests will fail. I suggest we proceed as follows:

- You leave the `hasattr` tests in for #716.
- I merge #718 before you merge #719 (preferred anyway, as it's easier to merge a smaller PR into a larger PR).
- You remove the `hasattr` tests in #719.

Then I think everything will work.

Reply:
OK, will do that.